### PR TITLE
Feature/UDS_Negative_Response_Draft1

### PR DESCRIPTION
**Pull request by AndreiDigit1**
I finished the first draft for Negative Response service for UDS.
The service contains:
- a custom message for every of six cases.
- a function for to send a frame.
- the responses are stored in unordered_map which include the negative response code and member from enum.
- it can be called by UDS services.

### CODE REVIEW

**Review by TheoRandstad — COMMENTED** [review]
You need to integrate into MCU and ECU modules.

File: src/uds/negative_response/include/negative_response.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#ifndef NEGATIVE_RESPONSE_H
```
Comment (Mihnea9901) [inline src/uds/negative_response/include/negative_response.h:1]: Files should be renamed like this: NegativeResponse.h, and test file should have a different name than the program file like NegativeResponseTest

Comment (TheoRandstad) [reply src/uds/negative_response/include/negative_response.h:1]: Test file seems to be same like the service, you need to use gtest for unit testing. See examples in test folder for ECU and MCU modules

**Review by TheoRandstad — APPROVED** [review]

**Review by TheoRandstad — COMMENTED** [review]
You also need to send the NRC back to the sender.(You use the same canID)

**Comment by AndreiDigit1** [comment]
> You also need to send the NRC back to the sender.(You use the same canID)

Thanks for the suggestion, I added a new commit.

**Pull request closed without merging**